src/main/java/bio/terra/service/filedata/flight/ingest/IngestBuildAndWriteScratchLoadFileStep.java

```diff
@@ -110,6 +109,58 @@ public StepResult doSkippableStep(FlightContext context) {
     }
   }
 
+  private void replaceJsonObject(
+      JsonNode node,
+      JsonNode fileRefNode,
+      String columnName,
+      Map<String, String> pathToFileIdMap,
+      AtomicLong failedRowCount) {
+    // replace
+    BulkLoadFileModel fileModel =
+        Objects.requireNonNull(objectMapper.convertValue(fileRefNode, BulkLoadFileModel.class));
+    String fileKey = fileModel.getTargetPath();
+    if (pathToFileIdMap.containsKey(fileKey)) {
+      String fileId = pathToFileIdMap.get(fileKey);
+      ((ObjectNode) node).put(columnName, fileId);
+    } else {
+      // If the file wasn't ingested, clear the node.
+      ((ObjectNode) node).removeAll();
+      failedRowCount.getAndIncrement();
+    }
+  }
+
+  private void replaceJsonArray(
+      JsonNode node,
+      JsonNode fileRefNode,
+      String columnName,
+      Map<String, String> pathToFileIdMap,
+      AtomicLong failedRowCount) {
+    List<BulkLoadFileModel> models =
+        StreamSupport.stream(Spliterators.spliteratorUnknownSize(fileRefNode.iterator(), 0), false)
+            .map(n -> objectMapper.convertValue(n, BulkLoadFileModel.class))
+            .collect(Collectors.toList());
+
+    long numNotIngested =
+        models.stream()
+            .filter(
+                model -> {
+                  String fileKey = model.getTargetPath();
+                  return !pathToFileIdMap.containsKey(fileKey);
+                })
+            .count();
+    if (numNotIngested > 0) {
```

Comment:
Since the count isn't needed, it's more efficient to use `allMatch()`
```suggestion
    boolean allIngested =
        models.stream()
            .allMatch(model -> pathToFileIdMap.containsKey(model.getTargetPath()));
    if (!allIngested) {
```

Reply:
It's really minor but right now the ordering of the `if` branches here doesn't match the ones in `replaceJsonObject()`.